codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,205 @@ public TextHeaders clear() {
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.add(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.add(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.addObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.addObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.setObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.setObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+    }
+
+    /**
+     * Will add multiple values for the same header as single header with a comma separated list of values.
+     *
+     * Please refer to section <a href="https://tools.ietf.org/html/rfc7230#section-3.2.2">3.2.2 Field Order</a>
+     * of RFC-7230 for details.
+     */
+    private final class SingleHeaderValuesComposer implements ValuesComposer {
+
+        private final ValueConverter<CharSequence> valueConverter;
+
+        SingleHeaderValuesComposer() {
+            valueConverter = valueConverter();
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            CharSequence current = DefaultTextHeaders.super.get(name);
+            if (current == null) {
+                DefaultTextHeaders.super.add(name, commaSeparated(value));
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparated(current, value));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            add(name, commaSeparated(values));
```

Comment:
In case this is the only header identified by `name` they `commaSeparated` may be called twice.  Depending on where we end up with the escape code this may escape the value twice.  Is this a problem, and can we avoid this situation?